mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -103,13 +109,106 @@ class MatrixFactorizationModel private[mllib] (
     recommend(productFeatures.lookup(product).head, userFeatures, num)
       .map(t => Rating(t._1, product, t._2))
 
+  /**
+   * Recommends topK users/products.
+   *
+   * @param num how many users to return. The number returned may be less than this.
+   * @return [Array[Rating]] objects, each of which contains a userID, the given productID and a
+   *  "score" in the rating field. Each represents one recommended user, and they are sorted
+   *  by score, decreasing. The first returned is the one predicted to be most strongly
+   *  recommended to the product. The score is an opaque value that indicates how strongly
+   *  recommended the user is.
+   */
+
+  /**
+   * Recommend topK products for all users
+   */
+  def recommendProductsForUsers(num: Int): RDD[(Int, Array[Rating])] = {
```

Comment:
Please check whether the return type is Java-friendly. You can generate the API doc and check the Java one.